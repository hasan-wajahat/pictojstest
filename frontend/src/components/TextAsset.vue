<template>
  <div class="text">
    <h4>Text</h4>
    <button @click="$emit('add-text', {type: 'text', text: ''})" class="btn btn-default">Add Text</button>
  </div>
</template>

<script>
export default {
  name: "TextAsset"
};
</script>