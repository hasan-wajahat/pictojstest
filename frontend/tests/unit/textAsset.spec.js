import { shallowMount } from '@vue/test-utils'
import TextAsset from '@/components/TextAsset.vue'

describe('TextAsset.vue', () => {
  it('renders text correctly', () => {
    const wrapper = shallowMount(TextAsset);
    expect(wrapper.text()).toMatchSnapshot();
  })
  
  it('button emits proper event', () => {
    const wrapper = shallowMount(TextAsset);
    wrapper.find('button').trigger('click');
    expect(wrapper.emitted()).toMatchSnapshot();
  })
})
