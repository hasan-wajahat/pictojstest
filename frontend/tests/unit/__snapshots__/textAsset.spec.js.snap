// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAsset.vue button emits proper event 1`] = `
Object {
  "add-text": Array [
    Array [
      Object {
        "text": "",
        "type": "text",
      },
    ],
  ],
}
`;

exports[`TextAsset.vue renders text correctly 1`] = `"Text Add Text"`;
